src/Compilers/VisualBasic/Portable/Binding/SemanticModelBinder.vb

```diff
@@ -8,15 +8,27 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
     Friend Class SemanticModelBinder
         Inherits Binder
 
-        Protected Sub New(containingBinder As Binder)
+        Private ReadOnly _ignoreAccessibility As Boolean = False
+
+        Protected Sub New(containingBinder As Binder, Optional ignoreAccessibility As Boolean = False)
             MyBase.New(containingBinder)
+
+            _ignoreAccessibility = ignoreAccessibility
         End Sub
 
-        Public Shared Function Mark(binder As Binder) As Binder
+        Public Shared Function Mark(binder As Binder, Optional ignoreAccessibility As Boolean = False) As Binder
             Return If(
                 binder.IsSemanticModelBinder,
```

Comment:
This condition is insufficient to ensure the existing binder is right. You also need to test its accessibility ignorance.